jaclang_jaseci/collections/base.py

```diff
@@ -0,0 +1,347 @@
+"""BaseCollection Interface."""
+
+from os import getenv
+from typing import (
+    Any,
+    AsyncGenerator,
+    Awaitable,
+    Generic,
+    Iterable,
+    Mapping,
+    Optional,
+    TypeVar,
+    Union,
+)
+
+from bson import ObjectId
+
+from motor.motor_asyncio import (
+    AsyncIOMotorClient,
+    AsyncIOMotorClientSession,
+    AsyncIOMotorCollection,
+    AsyncIOMotorCursor,
+    AsyncIOMotorDatabase,
+)
+
+from pymongo import (
+    DeleteMany,
+    DeleteOne,
+    IndexModel,
+    InsertOne,
+    UpdateMany,
+    UpdateOne,
+)
+from pymongo.server_api import ServerApi
+
+from ..utils import logger
+
+T = TypeVar("T")
+
+
+class BaseCollection(Generic[T]):
+    """
+    Base collection interface.
+
+    This interface use for connecting to mongodb.
+    """
+
+    __collection__: Optional[str] = None
+    __collection_obj__: Optional[AsyncIOMotorCollection] = None
+    __indexes__: list = []
+    __excluded__: list = []
+    __excluded_obj__: Optional[Mapping[str, Any]] = None
+
+    __client__: Optional[AsyncIOMotorClient] = None
+    __database__: Optional[AsyncIOMotorDatabase] = None
```

Comment:
Can you add a quick comment to each of these attributes documenting what they are? just like how you have described to me in the call